supabase/tests/live_specs_ext.test.sql

```diff
@@ -0,0 +1,85 @@
+-- We can resolve all roles granted with a minimum capability.
+-- This is commonly used for row-level security checks.
+create function tests.test_live_specs_ext()
+returns setof text as $$
+
+  -- Replace seed grants with fixtures for this test.
+  delete from user_grants;
+  insert into user_grants (user_id, object_role, capability) values
+    ('11111111-1111-1111-1111-111111111111', 'aliceCo/', 'admin'),
+    ('22222222-2222-2222-2222-222222222222', 'bobCo/', 'admin'),
+    ('33333333-3333-3333-3333-333333333333', 'carolCo/', 'read')
+  ;
+
+  delete from role_grants;
+  insert into role_grants (subject_role, object_role, capability) values
+    ('aliceCo/widgets/', 'bobCo/burgers/', 'admin'),
+    ('aliceCo/anvils/', 'carolCo/paper/', 'write'),
+    ('aliceCo/duplicate/', 'carolCo/paper/', 'read'),
+    ('aliceCo/stuff/', 'carolCo/shared/', 'read'),
+    ('carolCo/shared/', 'carolCo/hidden/', 'read')
+  ;
+
+  select results_eq(
+    $i$ select role_prefix::text, capability::text
+        from auth_roles()
+        order by role_prefix, capability
+    $i$,
+    $i$ VALUES  ('aliceCo/','admin'),
+                ('bobCo/burgers/','admin'),
+                ('carolCo/paper/','read'),
+                ('carolCo/paper/','write'),
+                ('carolCo/shared/', 'read')
+    $i$,
+    'alice roles'
+  );
+
+  select results_eq(
+    $i$ select role_prefix::text, capability::text from
+        internal.user_roles('22222222-2222-2222-2222-222222222222')
+        order by role_prefix, capability
+    $i$,
+    $i$ VALUES  ('bobCo/','admin')
+    $i$,
+    'bob roles'
+  );
+
+  select results_eq(
+    $i$ select role_prefix::text, capability::text from
+        internal.user_roles('33333333-3333-3333-3333-333333333333')
+        order by role_prefix, capability
+    $i$,
+    $i$ VALUES  ('carolCo/','read') $i$,
+    'carol roles'
+  );
+
+  -- seed live_specs, publication_specs, connectors and connector_tags 
+  delete from publications;
+  insert into publications (id, user_id) values
+    ('01:01:01:01:01:01:01:01', '11111111-1111-1111-1111-111111111111');
```

Comment:
optional: the `:` can be omitted, which I personally find a bit more readable: `'0101010101010101'`